api/src/wcs_api/services/beat_tracker.py

```
@@ -144,3 +144,82 @@ def track_beats(wav_path: str) -> BeatResult | None:
     if result is not None:
         return result
     return _track_with_librosa(wav_path)
+
+
+def detect_swing_ratio(wav_path: str, beats: list[float]) -> float | None:
+    """Estimate swung-vs-straight eighth-note feel from the onset
+    envelope.
+
+    For each beat interval, find where the onset energy peaks in the
+    offbeat region (0.40–0.80 of the interval). In straight time the
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)